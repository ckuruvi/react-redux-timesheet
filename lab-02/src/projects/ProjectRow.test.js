import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ProjectRow from './ProjectRow';

describe('<ProjectRow />', () => {
    let wrapper;

    beforeEach(() => {
        const project = {
            name: 'NAME',
            description: 'DESCRIPTION'
        };
        wrapper = shallow(<ProjectRow project={project} />);
    });

    it('should instantiate the project row component', () =>{
        expect(wrapper).toHaveLength(1)
    })

    it('should render values into expected calls', () => {
        expect(wrapper.find('td')).toHaveLength(2)
    })

    it('renders correctly', () => {
        const project = {
            name: 'NAME',
            description: 'DESCRIPTION'
        };
        const tree = renderer.create(<ProjectRow project={project} />).toJSON()
        expect(tree).toMatchSnapshot();
    })

});
