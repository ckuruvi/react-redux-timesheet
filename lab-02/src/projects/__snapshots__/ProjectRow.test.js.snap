// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectRow /> renders correctly 1`] = `
<tr>
  <td>
    NAME
  </td>
  <td>
    DESCRIPTION
  </td>
</tr>
`;
